manifests/client/install.pp

```diff
@@ -5,7 +5,19 @@
     fail 'class ::openldap::client has not been evaluated'
   }
 
-  package { $::openldap::client::package:
-    ensure => present,
+  case $::osfamily {
+    'Debian': {
+      package { $::openldap::client::package:
+        ensure => present,
+      }
+    }
+    'RedHat': {
+      class { '::openldap::client::utils':
```

Comment:
One issue I see with this is that currently there is no init.pp, but rather a client.pp and a server.pp. The issue that this PR is trying to address is that client.pp and server.pp both want to install the openldap-clients package on RedHat systems. That package resource needs to end up in the catalogue only once when openldap::client and openldap::server are both declared.

There are many ways we could address the duplicate resource. Some alternate solutions I can see are
1. Use ensure_packages/ensure_resource on the openldap-clients.
2. Make openldap::clients::utils not be included in server when on RedHat (or a parameter to bypass the include). It could still be declared directly via class/include, but would be mutually exclusive with openldap::client on RedHat.
3. Make openldap::client::utils completely decoupled from openldap::server. On Debian you could declare openldap::client, openldap::server, and openldap::client::utils as needed. On RedHat you could declare openldap::server and openldap::client or openldap::client::utils, but not both openldap::client and openldap::client::utils.
4. Expose an init.pp which could be used to configure the util_package. Have openldap::client include openldap::client::utils on RedHat. Have openldap::server include openldap::client::utils. Have openldap::client::utils use $openldap::utils_pacakge for the package resource.

I'm happy to code up an alternate implementation as time permits. Is adding an init.pp which provides a singular access to a parameter what you want?